src/propagate_constant.cpp

```diff
@@ -83,7 +84,12 @@ void propagate_constant::apply(module& m) const
     // Compute literals in parallel
     std::vector<instruction_ref> const_instrs_vec{const_instrs.begin(), const_instrs.end()};
     std::vector<argument> literals(const_instrs_vec.size());
-    simple_par_for(const_instrs_vec.size(), 1, [&](const auto i) {
+    std::size_t n = 1;
+#if !MIGRAPHX_HAS_EXECUTORS
+    n = std::max<std::size_t>(
```

Comment:
This is calculating the number of threads, it should be calculating the grainsize. You need to do something like `grainsize = const_instrs_vec.size() / n` to compute the grainsize.

Reply:
`grainsize = const_instrs_vec.size() / n`. If I understand correctly, we should rename `n` here to `grainsize`, what does `n` refer to in the calculation of `grainsize`?

Reply:
Discussed this over the call with Paul and here is what we would like it to be:
```
#if !MIGRAPHX_HAS_EXECUTORS
    std::size_t n = std::max<std::size_t>(2048 / std::thread::hardware_concurrency(), 1);
    grainsize = const_instrs_vec.size()/n;  
#endif
```
I will open up a PR and make the appropriate changes.

Reply:
PR opened with the changes: https://github.com/ROCm/AMDMIGraphX/pull/2993